creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.js

```diff
@@ -84,7 +84,7 @@ module.exports = async function (
       primary,
       secondary1,
       secondary2,
-      unhealthyReplicasSet,
+      unhealthyReplicasSet: new Set(unhealthyReplicas || []),
```

Comment:
what's the purpose of converting the `unhealthyReplicas` set into an array, only to convert it back to a set here? looks like validation just checks if the `unhealthyPeers` is an array rather than checking its contents

Reply:
it seems weird, right? but I have to do this because Bull serializes it in Redis, and then when it desererializes it's just an empty object (`{}`) without the Set context. so that was creating errors, and it's part of why I added that type validation (`_validateJobData()`) function